src/algokit/cli/init.py

```diff
@@ -165,55 +165,38 @@ def init_command(
-    # 1. it is slow to import on first execution after installing
-    # 2. the import fails if git is not installed
 
-    try:
-        # TODO: copier is typed, need to figure out how to force mypy to accept that or submit a PR
-        #       to their repo to include py.typed file
-        import copier  # type: ignore[import]
-    except ImportError as ex:
+    if not shutil.which("git"):
         raise click.ClickException(
             "Git not found; please install git and add to path.\n"
             "See https://github.com/git-guides/install-git for more information."
-        ) from ex
-
-    # TODO: in general, we should probably find a way to log all command invocations to the log file?
-    if template_name and template_url:
-        raise click.ClickException("Cannot specify both --template and --template-url")
+        )
 
     # parse the input early to prevent frustration - combined with some defaults but they can be overridden
     answers_dict = DEFAULT_ANSWERS | dict(answers)
 
+    template = _get_template(
+        name=template_name,
+        url=template_url,
+        commit=template_url_ref,
+        unsafe_security_accept_template_url=unsafe_security_accept_template_url,
+    )
+
     project_path = _get_project_path(directory_name)
     directory_name = project_path.name
     # provide the directory name as an answer to the template, if not explicitly overridden by user
     answers_dict.setdefault("project_name", directory_name)
 
-    if template_name:
-        blessed_templates = _get_blessed_templates()
-        template: TemplateSource = blessed_templates[template_name]
-    elif not template_url:
-        template = _get_template_url()
-    elif not _repo_url_is_valid(template_url):
-        logger.error(f"Couldn't parse repo URL {template_url}. Try prefixing it with git+ ?")
-        _fail_and_bail()
-    else:
-        logger.warning(_unofficial_template_warning)
-        # note: we use unsafe_ask here (and everywhere else) so we don't have to
-        # handle None returns for KeyboardInterrupt - click will handle these nicely enough for us
-        # at the root level
-        if not (
-            unsafe_security_accept_template_url
-            or questionary_extensions.prompt_confirm("Continue anyway?", default=False)
-        ):
-            _fail_and_bail()
-        template = TemplateSource(url=template_url, commit=template_url_ref)
-
     logger.debug(f"Attempting to initialise project in {project_path} from template {template}")
 
-    copier_worker = copier.run_copy(
+    # copier is lazy imported for two reasons
+    # 1. it is slow to import on first execution after installing
+    # 2. the import fails if git is not installed (which we check above)
+    # TODO: copier is typed, need to figure out how to force mypy to accept that or submit a PR
+    #       to their repo to include py.typed file
+    from copier.main import run_copy  # type: ignore[import]
```

Comment:
I moved this all the way down here, because this is a slow point anyway due to network fetch